### PR TITLE
feat(ui): app upload control

### PR DESCRIPTION
**Pull request by oshcherbakovv**

### CODE REVIEW

**Comment by oshcherbakovv** [comment]
Uploading Screen Recording 2024-12-27 at 01.15.41.mov…

**Comment by oshcherbakovv** [comment]
https://github.com/user-attachments/assets/a30f8229-4732-49d1-ad5b-1a922a97f3d2

File: ui/src/components/lib/file-input/file-input.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+export const FileInput = ({ onChange, onError, accept }: FileInputProps) => {
+  const { t } = useTranslation()
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const onDragOver = (event: DragEvent) => {
+    event.preventDefault()
+  }
+
+  const onChangeWithValidation = (files: FileList) => {
+    onError?.('')
```
Comment (DaniilSmirnov) [inline ui/src/components/lib/file-input/file-input.tsx:25]: maybe 'Validation Error' ?

File: ui/src/components/ui/device-control-panel/tabs/app-upload-control/app-upload-control.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    ApplicationInstallationService.name
+  )
+
+  return (
+    <>
+      <FileInput
+        accept={[
+          '.apk',
+          '.aab',
+          '.ipa',
```
Comment (DaniilSmirnov) [inline ui/src/components/ui/device-control-panel/tabs/app-upload-control/app-upload-control.tsx]: We can upload .ipa only for iOS devices (but it is not working yet)

File: ui/src/components/lib/file-input/file-input.tsx (hunk 1 above)
Comment (oshcherbakovv) [reply ui/src/components/lib/file-input/file-input.tsx:25]: This is just resetting the previous error text before starting the validation

File: ui/src/components/ui/device-control-panel/tabs/app-upload-control/app-upload-control.tsx (hunk 2 above)
Comment (oshcherbakovv) [reply ui/src/components/ui/device-control-panel/tabs/app-upload-control/app-upload-control.tsx]: Should I remove .ipa from this array for now?

Comment (DaniilSmirnov) [reply ui/src/components/ui/device-control-panel/tabs/app-upload-control/app-upload-control.tsx]: I think we dont need to accept ipa's on android devices

**Review by DaniilSmirnov — APPROVED** [review]

**Pull request merged**